x-pack/plugin/ent-search/src/test/java/org/elasticsearch/xpack/application/connector/syncjob/ConnectorSyncJobIndexServiceTests.java

```diff
@@ -887,6 +888,144 @@ public void testTransformConnectorFilteringToSyncJobRepresentation_WithFiltering
         assertEquals(connectorSyncJobIndexService.transformConnectorFilteringToSyncJobRepresentation(filtering), filtering1.getActive());
     }
 
+    public void testClaimConnectorSyncJob() throws Exception {
+        // Create sync job
+        PostConnectorSyncJobAction.Request syncJobRequest = ConnectorSyncJobTestUtils.getRandomPostConnectorSyncJobActionRequest(
+            connectorOneId
+        );
+        PostConnectorSyncJobAction.Response response = awaitPutConnectorSyncJob(syncJobRequest);
+        String syncJobId = response.getId();
+        Map<String, Object> syncJobSourceBeforeUpdate = getConnectorSyncJobSourceById(syncJobId);
+
+        @SuppressWarnings("unchecked")
+        Map<String, Object> syncJobConnectorBeforeUpdate = (Map<String, Object>) syncJobSourceBeforeUpdate.get(
+            ConnectorSyncJob.CONNECTOR_FIELD.getPreferredName()
+        );
+
+        // Claim sync job
+        ClaimConnectorSyncJobAction.Request claimRequest = new ClaimConnectorSyncJobAction.Request(
+            syncJobId,
+            randomAlphaOfLengthBetween(5, 100),
+            Map.of(randomAlphaOfLengthBetween(5, 100), randomAlphaOfLengthBetween(5, 100))
+        );
+        safeSleep(ONE_SECOND_IN_MILLIS);
```

Comment:
do we need to sleep, I don't think so? Immediate index refresh policy should allow you to get immediate changes reflected in the index, especially in unit test setup. 

Also, I feel `sleep` for 1s can quickly add up and result in slow unit tests